tools/wake/describe.cpp

```diff
@@ -169,16 +205,53 @@ static void describe_shell(const std::vector<JobReflection> &jobs, bool debug, b
   }
 }
 
-void describe(const std::vector<JobReflection> &jobs, bool script, bool debug, bool verbose,
-              const char *taguri) {
-  if (taguri) {
-    for (auto &job : jobs)
-      for (auto &tag : job.tags)
-        if (tag.uri == taguri) std::cout << tag.content << std::endl;
-  } else if (script) {
-    describe_shell(jobs, debug, verbose);
-  } else {
-    describe_human(jobs, debug, verbose);
+void describe_human(const std::vector<JobReflection> &jobs) {
+  for (auto &job : jobs) {
+    std::cout << "\n# " << job.label << "(" << job.job << ")\n$ ";
+    for (auto &cmd_part : job.commandline) {
+      std::cout << cmd_part << " ";  // # TODO: don't output trailing space
+    }
+    std::cout << "\n\n";
+
+    std::cout << "================================================\n" << std::endl;
+
+    // We have to use our speical stream for the output of the program
+    for (auto &log_line : job.std_writes) {
+      std::cout << log_line.first;
+    }
+
+    std::cout << "------------------------------------------------\n" << std::endl;
+  }
+}
+
+void describe(const std::vector<JobReflection> &jobs, DescribePolicy policy) {
+  switch (policy.type) {
+    case policy.SCRIPT: {
```

Comment:
Looking at this again I didn't realize that this was allowed syntax! Since SCRIPT is an enum from the *class* I wouldn't have expected you to be able to access it this way! TIL? I'm not sure what exactly I learned but I'm not aware of some corner case behavior that I wasn't before.

Reply:
Ah wait!!! I'm not crazy!

https://stackoverflow.com/questions/50027942/why-doesnt-clang-allow-accessing-a-nested-enum-class-through-an-instance

Please use DescribePolicy:SCRIPT instead